internal/vm/vsock.go

```diff
@@ -215,10 +209,15 @@ func tryConnect(logger *logrus.Entry, udsPath string, port uint32) (net.Conn, er
 		return nil, vsockConnectMsgError{cause: err}
 	}
 
-	err = tryConnRead(conn, vsockAckMsg(port), vsockAckMsgTimeout)
+	line, err := tryConnReadUntil(conn, '\n', vsockAckMsgTimeout)
 	if err != nil {
 		return nil, vsockAckError{cause: err}
 	}
+	if !strings.HasPrefix(line, "OK ") {
```

Comment:
Why not do the full string compare here?

Reply:
The line contains `assigned_hostside_port` (see https://github.com/firecracker-microvm/firecracker/blob/master/docs/vsock.md#examples) which we don't use.

We can check whether the following token is a number or not, but I'm unsure how much value we could get from doing so.

Reply:
Ahh, I thought this was the port we were passing in. Can we add the vsock.md link in the comments stating we are following a custom Firecracker protocol?

Reply:
Sure. Updated b67f582c9b34353cd8453820b0afeeab2d368016